server/game/core/ability/EventAbility.ts

```diff
@@ -17,7 +17,7 @@ export class EventAbility extends CardAbility {
     }
 
     public override meetsRequirements(context: AbilityContext = this.createContext(), ignoredRequirements = []) {
-        if (!ignoredRequirements.includes('location') && !this.isInValidLocation(context)) {
+        if (!ignoredRequirements.includes('location') && this.card.location !== Location.Discard) {
```

Comment:
Will this run into problems with the Set 3 card that lets you play events from discard?

Reply:
That's what `ignoredRequirements` is for, in that case we'd just include 'location' in the ignored requirements. We haven't done much play-from-out-of-play stuff yet so we still need to iron out the details but I know there's support for it in l5r so it's just a matter of learning the details